integration-tests/oidc-client-reactive/src/main/java/io/quarkus/it/keycloak/OidcClientRequestCustomFilter.java

```diff
@@ -0,0 +1,57 @@
+package io.quarkus.it.keycloak;
+
+import java.util.function.Consumer;
+
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.reactive.client.spi.ResteasyReactiveClientRequestContext;
+import org.jboss.resteasy.reactive.client.spi.ResteasyReactiveClientRequestFilter;
+
+import io.quarkus.oidc.client.Tokens;
+import io.quarkus.oidc.client.reactive.filter.OidcClientRequestReactiveFilter;
+import io.quarkus.oidc.client.runtime.AbstractTokensProducer;
+import io.quarkus.oidc.client.runtime.DisabledOidcClientException;
+import io.quarkus.oidc.common.runtime.OidcConstants;
+
+@Provider
+@Priority(Priorities.AUTHENTICATION)
+public class OidcClientRequestCustomFilter extends AbstractTokensProducer implements ResteasyReactiveClientRequestFilter {
+    private static final Logger LOG = Logger.getLogger(OidcClientRequestReactiveFilter.class);
+    private static final String BEARER_SCHEME_WITH_SPACE = OidcConstants.BEARER_SCHEME + " ";
+
+    protected void initTokens() {
+        if (earlyTokenAcquisition) {
+            LOG.debug("Token acquisition will be delayed until this filter is executed to avoid blocking an IO thread");
+        }
+    }
+
+    @Override
+    public void filter(ResteasyReactiveClientRequestContext requestContext) {
+        requestContext.suspend();
+
+        super.getTokens().subscribe().with(new Consumer<Tokens>() {
+            @Override
+            public void accept(Tokens tokens) {
+                requestContext.resume();
+                requestContext.getHeaders().add(HttpHeaders.AUTHORIZATION, BEARER_SCHEME_WITH_SPACE + tokens.getAccessToken());
```

Comment:
Actually, `resume()` should be the last thing to do as there is a small chance of a race condition. 

Same for error handler